src/uucore/src/lib/features/checksum.rs

```diff
@@ -710,6 +710,11 @@ fn identify_algo_name_and_length(
     }
 
     let bytes = if let Some(bitlen) = line_info.algo_bit_len {
+        if algorithm != ALGORITHM_OPTIONS_BLAKE2B {
+            // An invalid "bits" part was given to the regex
+            // eg: MD5-128 (foo.txt) = fffffffff
+            return None;
+        }
         if bitlen % 8 != 0 {
             // The given length is wrong
```

Comment:
Wouldn't it be easier to read if you check for the positive case? Something like
```rust
if algorithm == ALGORITHMS_OPTIONS_BLAKE2B && bitlen % 8 == 0 {
    Some(bitlen / 8)
} else {
    return None;
}
```

Reply:
I merged both conditions in a single if statement, but I find it is more understandable to check for the negative case. I've also slightly refactored the `identify_algo_name_and_length` function to make it return an error instead of an option. It helps making the difference between the error case and the case where no bit size was given.

WDYT ?